.github/workflows/build_docs.yml

```diff
@@ -0,0 +1,56 @@
+name: Build and deploy docs
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.10']
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install Dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install .[extra]
+        pip install .[doc]
+    - name: Build documentation
+      run: |
+        cd doc
+        make -C . html-no-examples
+    - name: Deploy Devs
+      if: success() && github.ref == 'refs/heads/master'
+      uses: JamesIves/github-pages-deploy-action@v4
+      with:
+        ssh-key: ${{ secrets.ACTIONS_DEPLOY_KEY }}
+        repository-name: dipy/preview-html
+        folder: ./doc/_build/html
+        target-folder: dev
```

Comment:
every PR or merge on master will override the last documentation build. is it the intended goal ?

Reply:
IMO, it should be. There is a `dev` API documentation version:
https://docs.dipy.org/dev/reference/index.html

So I assume that after every PR and as long as the doc build is successful, we want the `dev` version to be updated, otherwise, unless a manual deployment is done, the `dev` version will be the same as the last release, right?

Now, one thing to consider is that the examples are not being checked in this action. So maybe the deployment should be deferred to a later PR once we figure out whether there is a way to build the documentation with the examples within a reasonable time.